### PR TITLE
kripke: add cuda and rocm support

### PR DESCRIPTION
**Pull request by vsrana01**
This MR adds the SN-Neutronics proxy-app Kripke

### CODE REVIEW

**Comment by vsrana01** [comment]
@scheibelp - here is the MR for kripke

File: var/spack/repos/builtin/packages/kripke/package.py [hunk 1]
```diff
@@ -4,43 +4,95 @@
...
     """Kripke is a simple, scalable, 3D Sn deterministic particle
     transport proxy/mini app.
     """
 
     homepage = "https://computing.llnl.gov/projects/co-design/kripke"
     git = "https://github.com/LLNL/Kripke.git"
 
-    tags = ["proxy-app"]
-    version("1.2.4", submodules=True, tag="v1.2.4")
+    tags = ["SN-neutronics proxy-app"]
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/kripke/package.py]: Isn't this supposed to be two separate tags?  If so, should be a comma-separated list of two strings.

I suspect we don't support spaces within a tag.

**Review by tldahlgren — COMMENTED** [review]
Just commenting on tags atm.

File: var/spack/repos/builtin/packages/kripke/package.py (hunk 1 above)
Comment (rchen20) [inline var/spack/repos/builtin/packages/kripke/package.py]: Is `SN-neutronics` defined somewhere else? If not, shall we remove it?

File: var/spack/repos/builtin/packages/kripke/package.py [hunk 2]
```diff
@@ -4,43 +4,95 @@
...
     depends_on("mpi", when="+mpi")
-    depends_on("cmake@3.0:", type="build")
+    depends_on("blt")
+    depends_on("cmake")
     depends_on("caliper", when="+caliper")
+    depends_on("chai~examples+raja")
+    depends_on("raja~exercises~examples")
+    depends_on("umpire~examples")
+    depends_on("chai+raja+cuda", when="+cuda")
+    depends_on("raja+cuda", when="+cuda")
```
Comment (rchen20) [inline var/spack/repos/builtin/packages/kripke/package.py]: Is this redundant with respect to the line above?

File: var/spack/repos/builtin/packages/kripke/package.py [hunk 3]
```diff
@@ -4,43 +4,95 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 from spack.package import *
+import os
```
Comment (rchen20) [inline var/spack/repos/builtin/packages/kripke/package.py]: Maybe remove this? I don't see any calls in here requiring the `os` package.

**Review by rchen20 — DISMISSED** [review]

**Comment by scheibelp** [comment]
@spackbot help

**Comment by scheibelp** [comment]
@spackbot fix style

**Review by scheibelp — APPROVED** [review]

**Pull request merged**